gateway-ha/src/main/java/io/trino/gateway/ha/config/OAuth2GatewayCookieConfiguration.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.config;
+
+import com.google.common.collect.ImmutableSet;
+import io.airlift.units.Duration;
+
+import java.util.Set;
+
+public class OAuth2GatewayCookieConfiguration
+{
+    // Configuration initialization using dropwizard requires
+    // instance method setters. Values are global, and can be accessed using static getters
```

Comment:
I can revert this to the standard approach if desired. While working on this I have found that plumbing the configurations through `HaGatewayProviderModule` to be cumbersome and time consuming. Unlike Trino, where you can have multiple copies of a plugin with different configurations, all of the Gateway config is global. So I figured we could do away with the plumbing.

Reply:
I reworked this to use a singleton instead of statics, which looks better imo. LMK if you guys like the approach. I find the current way that configuration values are passed through constructors to be very cumbersome.

Please consider the naming. I decided on `*ConfigurationPropertiesProvider`, since it provides the values of the config properties of the corresponding `*Configuration` classes. Open to suggestions...